src/versionUtils.ts

```diff
@@ -47,39 +47,44 @@ export async function testVersionAvailable(versionToUse: string): Promise<boolea
 }
 
 export async function checkKamelNeedsUpdate(versionToUse?: string): Promise<boolean> {
-	const runtimeVersionSetting = vscode.workspace.getConfiguration().get(config.RUNTIME_VERSION_KEY) as string;
-	if (versionToUse) {
-		const passedVersionAvailable = await testVersionAvailable(versionToUse);
-		if (!passedVersionAvailable) {
-			const msg: string = `Camel K CLI Version ${versionToUse} unavailable.`;
-			extension.shareMessageInMainOutputChannel(msg);
-			return false;
-		}
-	} else if (!versionToUse) {
-		const latestversion = await getLatestCamelKVersion();
-		if (failed(latestversion)) {
-			console.error(latestversion.error);
-			return true;
+	const runtimeVersionSetting = config.getKamelRuntimeVersionConfig();
+	const autoUpgradeSetting: boolean | undefined = config.getKamelAutoupgradeConfig();
+	const kamelCliToolsPath: string | undefined = config.getActiveKamelconfig();
+	
+	if (autoUpgradeSetting || kamelCliToolsPath === undefined) {
+		if (versionToUse) {
+			const passedVersionAvailable = await testVersionAvailable(versionToUse);
+			if (!passedVersionAvailable) {
+				const msg: string = `Camel K CLI Version ${versionToUse} unavailable.`;
+				extension.shareMessageInMainOutputChannel(msg);
+				return false;
+			}
+		} else if (!versionToUse) {
+			const latestversion = await getLatestCamelKVersion();
+			if (failed(latestversion)) {
```

Comment:
I am not a friend of naming a function `failed`. It does say nothing about what the function does.

Reply:
This was [introduced a while ago](https://github.com/camel-tooling/vscode-camelk/pull/148) and not modified in this PR. i reported https://issues.redhat.com/browse/FUSETOOLS2-1175